app/jobs/custom_filter_records_count_update_job.rb

```diff
@@ -0,0 +1,14 @@
+class CustomFilterRecordsCountUpdateJob < ApplicationJob
+  queue_as :low
+
+  def perform
+    CustomFilter.find_each(batch_size: 25) do |filter|
+      Current.account = filter.account
+
+      next if filter.fetch_record_count_from_redis.nil?
+
+      filter.set_records_count
+      Current.reset
```

Comment:
Setting Current here needs improvement.  if the filter fails it might not be reset. rather make the account an argument for filter service that passing it on via current. 

also, move the filter calculation logic to a separate job as mentioned in above comment